shell/platform/fuchsia/runtime/dart/utils/mapped_resource.cc

```diff
@@ -0,0 +1,168 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "mapped_resource.h"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+#include <fuchsia/io/cpp/fidl.h>
+#include <fuchsia/mem/cpp/fidl.h>
+#include <lib/fdio/directory.h>
+#include <lib/fdio/io.h>
+#include <lib/syslog/global.h>
+#include <lib/trace/event.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <zircon/dlfcn.h>
+#include <zircon/status.h>
+
+#include "logging.h"
+#include "runtime/dart/utils/inlines.h"
+#include "runtime/dart/utils/logging.h"
+#include "runtime/dart/utils/vmo.h"
+#include "third_party/dart/runtime/include/dart_api.h"
+
+namespace dart_utils {
+
+static bool OpenVmo(fuchsia::mem::Buffer* resource_vmo,
+                    fdio_ns_t* namespc,
+                    const std::string& path,
+                    bool executable) {
+  TRACE_DURATION("dart", "LoadFromNamespace", "path", path);
+
+  // openat of a path with a leading '/' ignores the namespace fd.
+  dart_utils::Check(path[0] != '/', LOG_TAG);
+
+  if (namespc == nullptr) {
+    if (!VmoFromFilename(path, resource_vmo)) {
```

Comment:
Nit: Should this be `dart_utils::VmoFromFilename` for consistency?

Reply:
We're in the `dart_utils` namespace here, so I think the Google style guide allows (encourages?) using the unqualified name. At least, that's the convention we follow in Dart.